packages/dev/loaders/src/BVH/bvhLoader.ts

```diff
@@ -0,0 +1,391 @@
+import type { IAnimationKey } from "core/Animations";
+import { Animation } from "core/Animations/animation";
+import { Bone } from "core/Bones/bone";
+import { Skeleton } from "core/Bones/skeleton";
+import { Axis } from "core/Maths/math.axis";
+import { Matrix, Quaternion, Vector3 } from "core/Maths/math.vector";
+import type { Scene } from "core/scene";
+import type { Nullable } from "core/types";
+import type { BVHLoadingOptions } from "./bvhLoadingOptions";
+
+/**
+ * Original code from: https://github.com/herzig/BVHImporter/
+ *
+ * Ivo Herzig, 2016
+ * MIT License
+ */
+export class BVHParser {
+    private static readonly _XPosition = "Xposition";
+    private static readonly _YPosition = "Yposition";
+    private static readonly _ZPosition = "Zposition";
+    private static readonly _XRotation = "Xrotation";
+    private static readonly _YRotation = "Yrotation";
+    private static readonly _ZRotation = "Zrotation";
+
+    private static readonly _HierarchyNode = "HIERARCHY";
+    private static readonly _MotionNode = "MOTION";
+
+    /**
+     * Reads a BVH file, returns a skeleton
+     * @param text - The BVH file content
+     * @param scene - The scene to add the skeleton to
+     * @param loadingOptions - The loading options
+     * @returns The skeleton
+     */
+    public static ReadBvh(text: string, scene: Scene, loadingOptions: BVHLoadingOptions): Skeleton {
+        const lines = text.split("\n");
+
+        const animationName = loadingOptions.animationName ?? "Animation";
+        const loopBehavior = loadingOptions.loopBehavior ?? 1;
+        const skeletonName = loadingOptions.skeletonName ?? "skeleton";
+        const skeletonId = loadingOptions.skeletonId ?? "skeleton_id";
```

Comment:
The fallbacks are no longer needed since the loading options is required as are all the properties, yes? Seems like this whole block can simply be:
```suggestion
        const { animationName, loopBehavior, skeletonName, skeletonId } = loadingOptions;
```